rclcpp/test/rclcpp/test_node.cpp

```diff
@@ -162,6 +162,11 @@ TEST_F(TestNode, get_name_and_namespace) {
 }
 
 TEST_F(TestNode, subnode_get_name_and_namespace) {
+  {
+    auto node = std::make_shared<rclcpp::Node>("my_node", "ns");
+    EXPECT_STREQ("", node->get_sub_namespace().c_str());
+    EXPECT_STREQ("/ns", node->get_effective_namespace().c_str());
```

Comment:
So does this test block fail without your patch?

I was expecting a new test case that fails without the patch but passes with the new patch. It seems to me that this should already be working since you're not even creating a new sub node?

Reply:
It fails without the patch. As the only case when the sub_namespace was empty, so far, was when it was a normal node.

It does not throw or anything just the effective namespace for a normal node contains a trailing slash. That is why the second line will fail without the patch.

Currently the patch will allow sub_nodes with an empty sub_namespace,, ie. one that resides in the original namespace. This failed before, because of some checks that were done on the effective namespace, which previously contained the trailing slash and was considered invalid.

So if we want the old behaviour we would have to check for the empty sub_namespace explicitly now.